### DISCUSSION TITLE
Friendlier SQL: Function like lag/lead that offsets by range instead of row

### DISCUSSION THREAD

**Discussion by DanCory** [duckdb/duckdb#8697] [Ideas]
The lag and lead functions are quite useful for finding the previous or next row. But many calculations like year over year or week over week require finding the previous or next time unit.

This can be expressed in SQL using the RANGE clause with an expression like:
first_value(sales) over (order by year range between 1 following and 1 following)

This is similar to lag and lead which can be written using row between instead of range between.

We should be able to write:
lag_range(sales) over (order by year)

Here are some possible names:
- next and previous
- lag_range and lead_range
- lagger and leader
- trail and pilot
- before and after

It's hard to come up with something clearly different from lag and lead.

Similar to lag and lead, there should be an optional offset and default. Ignore Nulls would not make sense.

**Comment by Tishj** [comment]
I'm not sure I follow, we have the option to set a range for the frame of a window function.
https://duckdb.org/docs/sql/window_functions

You need to expand the `frame-spec` block to see it

**Comment by DanCory** [comment]
Thanks Tishj. I know we have the full window frame support and I use it. The blog was calling for "friendlier SQL" features and adding some shortcut functions for this would be useful.
